src/io/linux.zig

```diff
@@ -31,7 +31,24 @@ pub const IO = struct {
     pub fn init(entries: u12, flags: u32) !IO {
         // Detect the linux version to ensure that we support all io_uring ops used.
         const uts = std.os.uname();
-        const release = std.mem.sliceTo(&uts.release, 0);
+        const release = blk: {
+            // std.SemanticVersion requires there be no extra
+            // characters after the major/minor/patch numbers. But
+            // some distributions have extra characters, such as this
+            // Fedora one: 6.3.8-100.fc37.x86_64.
+            var last_valid_version_character_index: usize = 0;
+            for (uts.release) |c| {
+                if (c == '.' or (c >= '0' and c <= '9')) {
+                    last_valid_version_character_index += 1;
+                    continue;
+                }
```

Comment:
Believe this should handle the case of more than 3 dots (i.e. custom kernel) as `5.15.90.1-microsoft-standard-WSL2` fails to parse.

Reply:
Moved this into stdx.zig so I could add some unit tests.